flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -163,6 +135,39 @@ elideLengthsAlreadyInType(mlir::Type type, mlir::ValueRange lenParams) {
   return {};
 }
 
+/// Allocate a local variable.
```

Comment:
NIT: Is this unintended code movement resulting in diff ?

Reply:
> NIT: Is this unintended code movement resulting in diff ?

elideExtentsAlreadyInType, elideLengthsAlreadyInType are static functions and hence hence calls to these functions can occur only after its definition. allocateLocal is modified to call these functions and hence has to be moved after these functions.